WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -223,6 +241,11 @@ public bool TryGetSpentCoinByOutPoint(OutPoint outPoint, [NotNullWhen(true)] out
 			var toRemove = new List<SmartCoin>();
 			var toAdd = new List<SmartCoin>();
 
+			lock (Lock)
```

Comment:
A lock inside a lock is a deadlock

Reply:
Interestingly not in this case. Acquiring a lock in a context that already acquired the lock does nothing. Anyway, it's a mistake. I will address your comment during the weekend and detail why I think both caches are correct. My goal is to push atomic changes to reach #11516, a version of the `CoinsRegistry` much cleaner and easy to understand while keeping performances at least as high as with these 2 caches.